src/kbmod/reprojection_utils.py

```diff
@@ -38,28 +38,79 @@ def correct_parallax(coord, obstime, point_on_earth, guess_distance):
     # the object has an unknown distance from earth
     los_earth_obj = coord.transform_to(GCRS(obstime=obstime, obsgeoloc=loc))
 
-    cost = lambda d: np.abs(
-        guess_distance
-        - GCRS(ra=los_earth_obj.ra, dec=los_earth_obj.dec, distance=d * u.AU, obstime=obstime, obsgeoloc=loc)
+    cost = lambda geocentric_distance: np.abs(
+        heliocentric_distance
+        - GCRS(
+            ra=los_earth_obj.ra,
+            dec=los_earth_obj.dec,
+            distance=geocentric_distance * u.AU,
+            obstime=obstime,
+            obsgeoloc=loc,
+        )
         .transform_to(ICRS())
         .distance.to(u.AU)
         .value
     )
 
     fit = minimize(
         cost,
-        (guess_distance,),
+        (heliocentric_distance,),
     )
 
-    answer = GCRS(
-        ra=los_earth_obj.ra, dec=los_earth_obj.dec, distance=fit.x[0] * u.AU, obstime=obstime, obsgeoloc=loc
+    answer = SkyCoord(
+        ra=los_earth_obj.ra,
+        dec=los_earth_obj.dec,
+        distance=fit.x[0] * u.AU,
+        obstime=obstime,
+        obsgeoloc=loc,
+        frame="gcrs",
     ).transform_to(ICRS())
 
-    return answer
+    return answer, fit.x[0]
+
+
+def invert_correct_parallax(coord, obstime, point_on_earth, geocentric_distance, heliocentric_distance):
+    """Calculate the original ICRS coordinates of a point in EBD space, i.e. a result from `correct_parallax`.
+
+    Attributes
+    ----------
+    coord : `astropy.coordinate.SkyCoord`
+        The coordinate to be corrected for.
```

Comment:
I'm not really sure what "coordinate to be corrected for" means. Is that the input coordinate that is being modified? Or is that the pointing of the original image? Or is it something else?

Reply:
added some more clarification, it's an EBD coord that we're trying to find the original uncorrected position off